### DISCUSSION TITLE
HyperSPI with LolinS2-Mini change default pin-in

### DISCUSSION THREAD

**Discussion by d22b** [awawa-dev/HyperSPI#22] [Q&A] [answered]
Hi there,

actual i am building an SPi build to compare performance and optimize my board layout. I have everything ready in place, pre- testet and configured. I would like to drive an ws2812b stripe with an rasp4 via SPI. Because i saw different pictures of wiring i also want to doublecheck that.

1. Which connections do i need exactly from the rasp4 to the S2?
    actually wiring is:

gnd --> gnd
gpio8 --> 13
gpio11 --> 6
gpio9 --> 10
gpio10 -->  8

2. According to the wiring, i must change the default "pin in" on the S2. I read also the Tutorial: https://github.com/awawa-dev/HyperSerialEsp8266/wiki
   I found only the Information for data pinout in the platformio file. Can someone point me to the right configfile?

EDIT: The wiki describes, that the pin "can" be changed. That means i can use a different pin input without changing the firmware?

Thanks!

**Comment by awawa-dev** [comment]
Hi

> Which connections do i need exactly from the rasp4 to the S2?

I will post a picture of my esp32-s2 mini to Rpi Zero SPI wiring tomorrow.

Actually this the wiki for HyperSPI https://github.com/awawa-dev/HyperSPI/wiki and the data output pin is provided:
> 5 Change default data pin (-DDATA_PIN=2) to desired output.

you must compile a firmware using next steps.

The SPI input pins (the ones used to connect to the Rpi) can also be changed, but this requires changing one line in the source code, not platformio.ini. It is almost never required.

**Reply by d22b** [reply]
Thanks for the fast reply. Agree with your statement’s and I am aware of that.

„ The SPI input pins (the ones used to connect to the Rpi) can also be changed, but this requires changing one line in the source code, not platformio.ini. It is almost never required.“

Yes, that is exactly the point I am looking for. BTW I am using dedicated Wled board with fit PERFECTLY the needs for an multisegment setup. I can also change the line if you can point me to the file.

Great work.

**Comment by awawa-dev** [comment, marked as answer]
these are default SPI pins for ESP32-S2 mini

| ESP32-S2 lolin mini       | PINOUT    |
|---------------------------|-----------|
| Clock (SCK)               | GPIO 7    |
| Data (MOSI)               | GPIO 11   |
| SPI Chip Select(e.g. CE0) | GPIO 12   |
| GROUND                    | mandatory |
| LED output                | GPIO 2    |

if you want to change them you need to edit online this line:
https://github.com/awawa-dev/HyperSPI/blob/df40b76d7b719c177fad735ea0aeaf214dd4de82/src/main.cpp#L336

GPIO34 is a dummy pin, not used

**Reply by d22b** [reply]
Awesome. Thanks!

**Reply by LaytexLlama** [reply]
If you dont mind me asking, do you have an image of your setup? More specifically how you've connected you S2 mini to the Rpi. Happy to see you figured it out :)

**Reply by AstaRom** [reply]
> connected you S2 mini to the Rpi

![ESP32-S2 mini](https://user-images.githubusercontent.com/39379875/225410503-dfa575e0-2ecf-433a-98ac-258e47f11b2e.png)

**Reply by LaytexLlama** [reply]
Thanks heaps for this! One other question, do you have your S2 mini USB-C plugged into the Rpi also?

**Reply by AstaRom** [reply]
Right. S2 mini is powered by Rpi's Usb connector, while Rpi itself is powered by its own adapter.
